clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -158,9 +175,15 @@ public InetAddress currentAddress(String id) throws UnknownHostException {
      */
     public void disconnected(String id, long now) {
         NodeConnectionState nodeState = nodeState(id);
-        nodeState.state = ConnectionState.DISCONNECTED;
         nodeState.lastConnectAttemptMs = now;
-        updateReconnectBackoff(nodeState);
+        incrementReconnectBackoff(nodeState);
+        if (nodeState.state == ConnectionState.CONNECTING) {
+            incrementConnectionSetupTimeout(nodeState);
```

Comment:
`update` instead of `increment` here too?

Reply:
Refactored